internal/common/backoff/retry.go

```diff
@@ -123,3 +150,21 @@ Retry_Loop:
 		time.Sleep(next)
 	}
 }
+
+// ErrRetryableAfter returns a minimum delay until the next attempt.
+//
+// for most errors this will be 0, and normal backoff logic will determine
+// the full retry period, but e.g. service busy errors (or any case where the
+// server knows a "time until it is not useful to retry") are safe to assume
+// that a literally immediate retry is *not* going to be useful.
+//
+// note that this is only a minimum, however.  longer delays are assumed to
+// be equally valid.
+func ErrRetryableAfter(err error) (retryAfter time.Duration) {
```

Comment:
decided to move it here, and one thing needs it externally, so it's exposed.  and I kinda like the `backoff.ErrRetryableAfter` package/name, keeps it clear that it's retry-backoff-related.